src/main/webapp/js/search_bar.js

```diff
@@ -19,3 +19,131 @@ function searchToggle(obj, evt){
     }
 }
 
+
+/**
+ * Comparator function that compares two entities alphabetically
+ * @param a - a <td> tag object to compare the text of
+ * @param b - a <td> tag object to compare the text of
+ * @return  - positive if 'a' is larger than 'b'
+ *          - negative if 'a' is smaller than 'b'
+ *          - zero if 'a' equals 'b'
+ */
+function compareEntities(a, b) {
+    // Gets the text from the tag
+    var aText = $(a).text();
+    var bText = $(b).text();
+
+    // Returns numerical values based on the comparison of the texts
+    if (aText > bText) {
+        return 1;
+    } 
+    if (aText < bText) {
+        return -1;
+    }
+    return 0;
+}
+
+
+/**
+ * Comparator function that compares two entities chronologically by timestamps
+ * @param a - a <td> tag object to compare the text timestamps of
+ * @param b - a <td> tag object to compare the text timestamps of
+ * @return  - positive if 'a' is larger than 'b'
+ *          - negative if 'a' is smaller than 'b'
+ *          - zero if 'a' equals 'b'
+ */
+function compareTimestamps(a, b) {
+    // Gets the text and timestamp length
+    var aText = $(a).text();
+    var bText = $(b).text();
+    var aList = timestamps[aText];
+    var bList = timestamps[bText];
+
+    // Iterates through the lists up until the shorter length, checking the values in order
+    var limit = (aList.length > bList.length) ? bList.length : aList.length;
+    for (var i = 0; i < limit; i++) {
+        if (aList[i] > bList[i]) {
+            return 1;
+        }
+        if (aList[i] < bList[i]) {
+            return -1;
+        } 
+    }
+
+    // If all contents are equal up until the limit, the shorter list will preceed the longer list
+    if (aList.length > bList.length) {
+        return 1;
+    }
+    if (aList.length < bList.length) {
+        return -1;
+    }
+
+    // If the lists are equal, we sort alphabetically 
+    return compareEntities(a, b);
+}
+
+
+/**
+ * Filters the entities based on the current text of the searchbar
+ */
+function filterEntities() {
+    var query = $('#entity-seachbar').val().toLowerCase();
+    var entities = $('#output table tr td:first-child');
+
+    // Case 1: Searchbar is empty, so all entities are displayed
+    if (query == '') {
```

Comment:
Let's do query.trim() == '' here since users could enter empty spaces

Reply:
change reflected in newest commit